js/string/string.js

```diff
@@ -88,7 +88,7 @@ var string = {
 	getObject: function (name, roots, add) {
 		// The parts of the name we are looking up
 		// `['App','Models','Recipe']`
-		var parts = name ? name.split('.') : [],
+		var parts = "" + name ? ("" + name).split('.') : [],
```

Comment:
Looks good.  Thinking about this, it's possible, but unlikely, that someone could try to get the `""` property.

```
var obj = {"": "empty"};
getObject("",[obj]);
```

I think we should account for this.